src/sentry/integrations/msteams/webhook.py

```diff
@@ -180,3 +199,60 @@ def handle_member_removed(self, request):
 
         integration.delete()
         return self.respond(status=204)
+
+    def handle_action_submitted(self, request):
+        data = request.data
+        channel_data = data["channelData"]
+        team_id = channel_data["team"]["id"]
+        tenant_id = channel_data["tenant"]["id"]
+        group_id = data["value"]["groupId"]
+        user_id = data["from"]["id"]
+
+        try:
+            integration = Integration.objects.get(provider=self.provider, external_id=team_id)
+        except Integration.DoesNotExist:
+            logger.info("msteams.action.missing-integration", extra={"team_id": team_id})
+            return self.respond(status=404)
+
+        try:
+            group = Group.objects.select_related("project__organization").get(
+                project__in=Project.objects.filter(
+                    organization__in=integration.organizations.all()
+                ),
+                id=group_id,
+            )
+        except Group.DoesNotExist:
+            logger.info(
+                "msteams.action.invalid-issue",
+                extra={"team_id": team_id, "integration_id": integration.id},
+            )
+            return self.respond(status=404)
+
+        try:
+            idp = IdentityProvider.objects.get(type="msteams", external_id=team_id)
+        except IdentityProvider.DoesNotExist:
+            logger.info(
+                "msteams.action.invalid-team-id",
+                extra={
+                    "team_id": team_id,
+                    "integration_id": integration.id,
+                    "organization_id": group.organization.id,
+                },
+            )
+            return self.respond(status=404)
+
+        try:
+            Identity.objects.select_related("user").get(idp=idp, external_id=user_id)
```

Comment:
Can we change this to an exists() query please like I mentioned? And drop the select_related? It makes this query more complex for what it’s doing.

Reply:
My bad, I totally forgot about that. Will fix.

Reply:
No worries, wanted to make sure it wasn’t lost.

For context, the exists() translates to a simpler/faster query on the database when you’re only checking for existence like you are here. It doesn’t need to return the row data. Dropping the select_related also drops the JOIN on the auth_user table, which also isn’t needed here since we don’t need any row data.

Reply:
Actually one thing I realized is that we will want to use the result of this call later (once I implement users doing actions in Sentry through Teams messages). I only changed it like this because flake8 was complaining about unused variables. Given that, do you think I should still change it, or no?

Reply:
If you plan on using it later that’s fine. Maybe a quick comment added would have clarified that, or doing `_ = Identity...` would have appeased the linter. But yeah, I think just explaining that is better than changing the logic here for something temporary.